File: python/src/Structural/Flyweight/TeaFlavor.py

#//TeaFlavor.java - the Concrete Flyweight

from TeaOrder import TeaOrder

class TeaFlavor(TeaOrder) :
	def __init__(self,teaFlavor) :
		self.teaFlavor = teaFlavor

	def getFlavor(self) : 
		return self.teaFlavor

	def serveTea(self,teaOrderContext) :
		return "Serving tea flavor {0} to table number {1}".format( self.teaFlavor, teaOrderContext.getTable())

File: csharp/src/Structural/Flyweight/TeaFlavor.cs
namespace Structural.Flyweight {


//TeaFlavor.java - the Concrete Flyweight

public class TeaFlavor : TeaOrder {  
    string teaFlavor; 
   
    public TeaFlavor(string teaFlavor) {
        this.teaFlavor = teaFlavor;
    }
   
    public string getFlavor() {
        return this.teaFlavor;
    }
   
    override public void serveTea(TeaOrderContext teaOrderContext) {
        //Console.WriteLine("Serving tea flavor " + teaFlavor + " to table number " + teaOrderContext.getTable());
    }
}

}
